### PR TITLE
feat: add report command

### PR DESCRIPTION
**Pull request by tooboredtocode**

### CODE REVIEW

**Comment by tooboredtocode** [comment]
ok the report command was added by accident

File: bot/cogs/utility/report.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+            return
+
+        # this sends the embed to the selected channel
+        embed = discord.Embed(title="Reported Message:", color=15158332)
+        embed.add_field(name="Reported Message Author", value=f"{reported_msg.author.mention}")
+        if len(reported_msg.content) < 1024:
+            embed.add_field(name="Reported Message Content", value=f"{reported_msg.content}")
+        else:
+            embed.add_field(name="Reported Message Content (1)", value=f"{reported_msg.content[:1024]}")
+            embed.add_field(name="Reported Message Content (2)", value=f"{reported_msg.content[1025:]}")
```
Comment (JMdoubleU) [inline bot/cogs/utility/report.py:37]: The 1024 bound above is exclusive, so this should be `1024:`

File: bot/cogs/utility/report.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+                embed.add_field(name="Reported Message Content", value=f"{field_reported_msg_content}")
+            else:
+                embed.add_field(name="Reported Message Content (1)", value=f"{field_reported_msg_content_1}")
+                embed.add_field(name="Reported Message Content (2)", value=f"{field_reported_msg_content_2}")
+            embed.add_field(name="Reported by:", value=f"{field_reporter}")
+            embed.add_field(name="Link to message:", value=f"{field_url}")
+            await msg.edit(embed=embed)
+            await msg.clear_reactions()
+            pass
+            return
```
Comment (JMdoubleU) [inline bot/cogs/utility/report.py:90]: `pass` and `return`???

File: bot/cogs/utility/report.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
+import discord
+from discord.ext import commands
+import asyncio
+
+from bot.bot import Bot
+from config.config import report_channel_id as report_channel_id
+from config.config import report_accept_channel_id as accept_channel_id
+
+
+class report(commands.Cog):
```
Comment (JMdoubleU) [inline bot/cogs/utility/report.py:10]: nitpick: [PEP 8](https://www.python.org/dev/peps/pep-0008/#class-names)

File: bot/cogs/utility/nickrequest.py [hunk 4]
```diff
@@ -21,12 +21,10 @@ def __init__(self, bot: Bot):
     async def nick(self, ctx: commands.Context, *, nick):
         acc_channel = self.bot.get_channel(accept_channel_id)
         nick = str(nick)
-        nick.replace('"', '\"')
-        nick.replace("'", "\'")
 
         # this looks if the request is valid
         for char in nick:
-            if not char in allowed_list:
+            if not char in string.printable:
```
Comment (JMdoubleU) [inline bot/cogs/utility/nickrequest.py]: Probably doesn't matter at this scale but it'd be better to reuse a `set(string.printable)` for this.

File: bot/cogs/utility/report.py [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+        field_reported_msg_author = easy_embed['fields'][0]['value']
+        if easy_embed['fields'][1]['name'] == "Reported Message Content":
+            field_reported_msg_content = easy_embed['fields'][1]['value']
+        else:
+            field_reported_msg_content_1 = easy_embed['fields'][1]['value']
+            field_reported_msg_content_2 = easy_embed['fields'][2]['value']
+        field_reporter = easy_embed['fields'][-2]['value']
+        field_url = easy_embed['fields'][-1]['value']
+
+        rp_channel = server.get_channel(int(field_url[-37:-19]))
```
Comment (JMdoubleU) [inline bot/cogs/utility/report.py]: gg if these ID sizes ever change, would be better to split the path by `/`

File: bot/cogs/utility/report.py (hunk 2 above)
Comment (tooboredtocode) [reply bot/cogs/utility/report.py:90]: Yes, as I want the function to end if an exception occurs as well as avoid clogging up the log.

Comment (JMdoubleU) [reply bot/cogs/utility/report.py:90]: It should just be `return`, `pass` is a no-op statement

Comment (tooboredtocode) [reply bot/cogs/utility/report.py:90]: 🤦 I guess i'm way too tired

**Comment by tooboredtocode** [comment]
Im sorry, this will be the last unclean pr

**Pull request closed without merging**